VolumeWidget/ArticleWidget/ArticleWidgetProvider.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  ArticleWidgetProvider.swift
+//  Volume
+//
+//  Created by Vin Bui on 11/11/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Combine
+import WidgetKit
+
+struct ArticleWidgetProvider: TimelineProvider {
+
+    // MARK: - Properties
+
+    private let articlesLimit: Double = 4
+
+    class ProviderCancellable {
+        static var queryBag = Set<AnyCancellable>()
+    }
+
+    // MARK: - TimelineProvider Methods
+
+    /// Provides an Article entry representing a placeholder version of the Article widget.
+    func placeholder(in context: Context) -> ArticleEntry {
+        ArticleEntry(date: .now, article: ArticleWidgetProvider.dummyArticle!)
```

Comment:
Why not make `dummyArticle` None optional since its a constant and avoid force casting

Reply:
That's because `dummyArticle` is created with Apollo. In order to create an `Article` object, you must create an `ArticleQuery` given some `[String: Any]` dictionary. This must be marked with `try` which will return `nil` if the decoding fails.